### PR TITLE
CUMULUS-293 schema validation

### PR DESCRIPTION
**Pull request by sethvincent**
- adds schema validation
- adds tests for schema validation
- adds `schemas` param to `loadNestedEvent` and `createNextEvent` as expected by `__main__.py` and the python adapter
- adds schemas to input for ` __main__.py` as expected by the java & js adapters

### CODE REVIEW

File: message_adapter/message_adapter.py [hunk 1]
```diff
@@ -154,6 +155,13 @@ def __loadConfig(self, event, context):
 
         raise LookupError('Unknown event source: ' + source)
 
+    def __validate_json(self, document, schema_filepath):
```
Comment (laurenfrederick) [inline message_adapter/message_adapter.py]: Is it possible to preface the error messages with which schema is failing?

Comment (sethvincent) [reply message_adapter/message_adapter.py]: Good idea

File: __main__.py [hunk 2]
```diff
@@ -6,31 +6,35 @@
...
-      result = transformer.loadNestedEvent(event, context)
-    elif (functionName == 'createNextEvent'):
-      handlerResponse = allInput['handler_response']
-      if 'message_config' in allInput:
-        messageConfig = allInput['message_config']
-      else:
-        messageConfig = None
-      result = transformer.createNextEvent(handlerResponse, event, messageConfig)
+    functionName = sys.argv[1]
+    transformer = message_adapter.message_adapter()
```
Comment (jennyhliu) [inline __main__.py]: transformer = message_adapter.message_adapter()
This could take the schemas from the input message if exists.

Comment (sethvincent) [reply __main__.py]: :+1: Thanks I agree, that seems like a good idea.

**Comment by sethvincent** [comment]
@laurenfrederick I updated CONTRACT.md with some details about the json schemas.

**Pull request merged**